server/src/main/java/org/elasticsearch/cluster/metadata/DataStreamLifecycle.java

```diff
@@ -331,7 +340,8 @@ public XContentBuilder toXContent(
         XContentBuilder builder,
         Params params,
         @Nullable RolloverConfiguration rolloverConfiguration,
-        @Nullable DataStreamGlobalRetention globalRetention
+        @Nullable DataStreamGlobalRetention globalRetention,
+        boolean isInternalDataStream
```

Comment:
Maybe worth mentioning that we never deserialize the thing that is serialized here? That threw me off at first. The other `toXContent` is for serialization to the cluster state, and this one (other than when called by that one) is just for serializing for user output. Also, it doesn't look like this one is tested outside of the `null, null, false` args passed by the other one?

Reply:
I will add a comments to clarify this.

We have dedicated tests like this one: https://github.com/elastic/elasticsearch/blob/42706c62a6e2086c666c0b510520127ff9777d14/server/src/test/java/org/elasticsearch/cluster/metadata/DataStreamLifecycleTests.java#L112 in a few places that use it. Did you have something else in mind?